backend/src/controllers/stockController.ts

```diff
@@ -1,129 +1,254 @@
+//             sendError(res, err as CustomError);
+//         } else {
+//             sendError(res, err as CustomError);
+//         }
+//     }
+// };
+//
+// export const deleteStockItem = async (req: Request, res: Response, connection: PoolConnection, stockID: number, itemID: number) => {
+//     try {
+//         const stockService = new StockService(connection);
+//         const result = await stockService.deleteStockItem(stockID, itemID);
+//         if (result.affectedRows === 0)
+//             sendError(res, new NotFoundError("Item not found or already deleted.", ErrorMessages.DeleteStockItem));
+//         res.status(200).json({message: "Stock item deleted successfully."});
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
+//
+// export const getAllItems = async (req: Request, res: Response, connection: PoolConnection) => {
+//     try {
+//         const stockService = new StockService(connection);
+//         const items = await stockService.getAllItems();
+//         res.status(200).json(items);
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
+//
+// export const getItemDetails = async (req: Request, res: Response, connection: PoolConnection, itemID: number) => {
+//     try {
+//         const stockService = new StockService(connection);
+//         const items = await stockService.getItemDetails(itemID);
+//         if (items.length === 0)
+//             sendError(res, new NotFoundError("Item not found.", ErrorMessages.GetItemDetails));
+//         res.json(items[0]);
+//     } catch (err: any) {
+//         sendError(res, err as CustomError);
+//     }
+// };
+export class StockController {
+    private stockService: StockService;
 
-export const getAllStocks = async (req: Request, res: Response, connection: PoolConnection) => {
-    try {
-        const stockService = new StockService(connection);
-        const stocks = await stockService.getAllStocks();
-        res.status(200).json(stocks);
-    } catch (err: any) {
-        sendError(res, ErrorMessages.GetAllStocks, err);
+    constructor(
+        readStock: ReadStockRepository,
+        writeStock: WriteStockRepository
+    ) {
+        this.stockService = new StockService(readStock, writeStock);
     }
-};
 
-export const createStock = async (req: Request, res: Response, connection: PoolConnection) => {
-    try {
-        const stock: Partial<StockToCreate> = extractDataFromRequestBody(req, ['LABEL', 'DESCRIPTION']);
-        const stockService = new StockService(connection);
-        await stockService.createStock( stock);
-        res.json({message: "Stock created successfully."});
-    } catch (err: any) {
-        sendError(res, ErrorMessages.CreateStock, err);
+   public async getAllStocks(req: Request, res: Response) {
+        try {
+            const stocks = await this.stockService.getAllStocks();
+            res.status(200).json(stocks);
```

Comment:
suggestion: you could use constant for http codes at module level to avoid magic number.

```
const HTTP_CODE_OK = 200
```